api-v2/src/student/dto/create-student.dto.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  IsEmail,
+  IsNotEmpty,
+  IsInt,
+  IsString,
+  Max,
+  Min,
+  IsOptional,
+  IsObject,
+} from 'class-validator';
+import { ScheduleCourse } from '../../../../common/types';
+
+export class CreateStudentDto {
+  @IsNotEmpty()
+  @IsString()
+  fullName: string;
+
+  @IsString()
+  nuid: string;
+
+  @IsNotEmpty()
+  @IsEmail()
+  email: string;
+
+  @IsNotEmpty()
+  @IsString()
+  password: string;
+
+  @IsOptional()
+  @IsInt()
+  @Min(1898) // the year NEU was established
+  @Max(3000) // will GraduateNU last beyond year 3000?!
+  academicYear?: number;
```

Comment:
I am little concerned or I am just thinking if we should make academic graduateYear and stuff optional here. I know that for our current implementation a student signs up first, and when that happens these fields are all optional. I am just a bit concerned that if frontend doesn't do a good job enforcing onboarding, then these values will be null, and it will potentially break the home plan page. Any thoughts on this?

Reply:
i talked with cindy on slack a bit about this and i think in the future we should enforce onboarding on the frontend, and always assume user has graduation date on the backend (make it required)

for the case where onboarding gets interrupted (user reloads page/quits browser), we should save in localstorage between steps in the browser, and resume on frontend if necessary